src/internal/SlidingSheet.js

```diff
@@ -0,0 +1,439 @@
+     *
+     * @param {boolean} open If true, the `SlidingSheet` was requested to be opened.
+     * @param {string} reason The reason for the open or close request. Possible values are
+     * 'swipe' for open requests; 'clickaway' (on overlay clicks),
+     * 'escape' (on escape key press), and 'swipe' for close requests.
+     */
+    onRequestChange: PropTypes.func.isRequired,
+    /**
+     * If true, the `SlidingSheet` is opened.
+     */
+    open: PropTypes.bool.isRequired,
+    /**
+     * The CSS class name to add to the `Overlay` component that is rendered behind the `SlidingSheet`.
+     */
+    overlayClassName: PropTypes.string,
+    /**
+     * Override the inline-styles of the `Overlay` component that is rendered behind the `SlidingSheet`.
+     */
+    overlayStyle: PropTypes.object,
+    /**
+     * The width or height of the `SlidingSheet`, in the direction in which it should slide.
+     */
+    primaryDimension: PropTypes.number,
+    /**
+     * Override the inline-styles of the root element.
+     */
+    style: PropTypes.object,
+    /**
+     * The width of the area in pixels on the appropriate edge of the screen where the `SlidingSheet`
+     * can be swiped open from. Setting this to `null` spans that area to the entire page
+     * (**CAUTION!** Setting this property to `null` might cause issues with sliders and swipeable
+     * `Tabs`, and for vertical sheets, will interfere with scrolling: use at your own risk).
+     */
+    swipeAreaWidth: PropTypes.number,
+    /**
+     * If true, swiping in the opposite of direction will open the `SlidingSheet`.
+     */
+    swipeToOpen: PropTypes.bool,
+    /**
+     * The zDepth of the `SlidingSheet`.
+     */
+    zDepth: propTypes.zDepth,
+
+  };
+
+  static defaultProps = {
+    closeable: true,
+    direction: 'left',
+    modal: false,
+    swipeAreaWidth: 30,
+    swipeToOpen: false,
+    zDepth: 2,
+  };
+
+  static contextTypes = {
+    muiTheme: PropTypes.object.isRequired,
+  };
+
+  componentWillMount() {
+    this.maybeSwiping = false;
+    this.touchStartX = null;
+    this.touchStartY = null;
+    this.swipeStartPrimary = null;
+
+    this.setState({
+      swiping: null,
+    });
+  }
+
+  componentDidMount() {
+    this.enableSwipeHandling();
+  }
+
+  componentDidUpdate() {
+    this.enableSwipeHandling();
+  }
+
+  componentWillUnmount() {
+    this.disableSwipeHandling();
+  }
+
+  getStyles() {
+    const muiTheme = this.context.muiTheme;
+    const vertical = this.isVertical();
+
+    const x = this.getTranslateMultiplier() * ((this.props.open || vertical) ? 0 : this.getMaxTranslatePrimary());
+    const y = this.getTranslateMultiplier() * ((this.props.open || !vertical) ? 0 : this.getMaxTranslatePrimary());
+
+    const height = vertical ? this.props.primaryDimension : '100%';
+    const width = vertical ? '100%' : this.props.primaryDimension;
+
+    const styles = {
+      root: {
```

Comment:
this style is not applied on the `root`. We could use a different name, like paper?